unity-renderer/Assets/DCLPlugins/ECS7/Systems/PlayerSystem/ECSPlayerTransformSystem.cs

```diff
@@ -63,21 +71,40 @@ public void Update()
             {
                 scene = loadedScenes[i];
 
-                var transform = TransformHelper.SetTransform(scene, ref avatarPosition, ref avatarRotation, ref currentWorldOffset);
-
-                componentsWriter.PutComponent(scene.sceneData.sceneNumber, SpecialEntityId.PLAYER_ENTITY, ComponentID.TRANSFORM,
-                    transform, lamportTimestamp, ECSComponentWriteType.SEND_TO_SCENE);
-
-                componentsWriter.PutComponent(scene.sceneData.sceneNumber, SpecialEntityId.INTERNAL_PLAYER_ENTITY_REPRESENTATION, ComponentID.TRANSFORM,
-                    transform, ECSComponentWriteType.EXECUTE_LOCALLY);
+                if (!componentsWriter.TryGetValue(scene.sceneData.sceneNumber, out var writer))
+                    continue;
+
+                var pooledTransform = transformPool.GetElement();
+                var transform = pooledTransform.WrappedComponent.Model;
+                transform.position = SetInSceneOffset(scene, ref avatarPosition, ref currentWorldOffset);
+                transform.rotation = avatarRotation;
+
+                writer.Put(SpecialEntityId.PLAYER_ENTITY, ComponentID.TRANSFORM, pooledTransform);
+
+                if (scene.entities.TryGetValue(SpecialEntityId.PLAYER_ENTITY, out var entity))
+                {
+                    ECSTransform stored = transformComponent.Get(scene, entity).model ?? new ECSTransform();
+                    stored.position = transform.position;
+                    stored.rotation = transform.rotation;
+                    stored.scale = transform.scale;
+                    stored.parentId = transform.parentId;
+                    transformComponent.SetModel(scene, entity, stored);
+                }
             }
-
-            lamportTimestamp++;
         }
 
         private void OnSceneLoaded(IParcelScene obj)
         {
             newSceneAdded = true;
         }
+
+        private static Vector3 SetInSceneOffset(IParcelScene scene, ref Vector3 position, ref Vector3 worldOffset)
+        {
+            Vector3 offsetposition = new Vector3();
+            offsetposition.x = (position.x + worldOffset.x) - scene.sceneData.basePosition.x * ParcelSettings.PARCEL_SIZE;
+            offsetposition.y = (position.y + worldOffset.y);
+            offsetposition.z = (position.z + worldOffset.z) - scene.sceneData.basePosition.y * ParcelSettings.PARCEL_SIZE;
+            return offsetposition;
```

Comment:
I see it second time. Probably can be extracted as utility or extension method (that will set up offset already in the Vector3 constructor)

Reply:
done, moved to a utility class